docs/design/job_architecture.md

```diff
@@ -199,72 +87,78 @@ The actual message that the JobComm sees in the kernel has this format:
 ```
 All messages have a `request_type`, most specify a job or list of job IDs, and a few have some extra info.
 
-## Messages sent to the kernel
-These are organized by the `request_type` field, followed by the expected response message. Additional parameters and their formats are given as a list below the request name. The job params and message type terms are imported into the JobComm module as PARAM and MESSAGE_TYPE respectively, i.e.
+The documentation will use the message type and parameter names used in the job config file (e.g. `STATUS` instead of `job_status`, `RETRY` instead of `retry_job`, `JOB_ID` instead of `job_id`, etc.) as the JS code will refer to those values using the names in the JobCommMessages object, rather than hardcoding the strings.
 
-```py
-# in biokbase.narrative.jobs.jobcomm
-(PARAM, MESSAGE_TYPE) = load_job_constants()
+### `CANCEL`
 
-print(MESSAGE_TYPE["STATUS"]) # prints 'job_status'
-print(PARAM["JOB_ID_LIST"])   # prints 'job_id_list'
-```
+Request that the server cancel the running job(s). If the job ID supplied is a batch ID, EE2 will cancel the job and all its child jobs.
 
-For simplicity below, it is assumed that constants have been created from the key/value pairs in MESSAGE_TYPE and PARAM.
+**Arguments**
+  * `JOB_ID` - a string, the job id OR
+  * `JOB_ID_LIST` - an array of job IDs
 
-An example of a job status message sent from the frontend to the kernel:
-```json
-{
-  "msg_id": "some string",
-  "content": {
-    "data": {
-      "request_type": "job_status",
-      "job_id": "a_job_id",
-    }
-  }
-}
-```
-Messages expected by the narrative backend:
+### `INFO`
 
-`STATUS` - request job status
+Request information about the job(s), specifically app id, spec, input parameters and (if finished) outputs
+
+**Arguments**
   * `JOB_ID` - a string, the job id OR
   * `JOB_ID_LIST` - an array of job IDs OR
   * `BATCH_ID` - a batch parent (make the request for all jobs in the batch)
 
-`STATUS_ALL` - request the status of all jobs registered in the job manager
+### `LOGS`
+
+Request the job logs starting at some given line. Note that the `first_line`, `num_lines`, and `latest` arguments apply to all jobs in the `JOB_ID_LIST`
+
+**Arguments**
+  * `JOB_ID` - a string, the job id OR
+  * `JOB_ID_LIST` - an array of job IDs
+  * `first_line` - the first line (0-indexed) to request (optional)
+  * `num_lines` - the number of lines to request (will get back up to that many if there aren't more) (optional)
+  * `latest` -  true if requesting just the latest set of logs (optional)

```

Comment:
No, but that can be easily remedied.

Originally the JCM module came about because the FE and BE were using different sets of terms for the same information (the backend was using snake case and the FE camel case), and the JobCommChannel translated every message from FE vocab to BE and vice versa. The job logs params must have been unified earlier, or perhaps they were never camel-cased in the FE? Anyway, now fixed.

Reply:
:+1: